warehouse/warehouse.go

```diff
@@ -725,6 +729,59 @@ func (wh *HandleT) runUploadJobAllocator() {
 	}
 }
 
+func (wh *HandleT) uploadStatusTrack() {
+	for {
+		time.Sleep(uploadStatusTrackFrequency)
+
+		for _, warehouse := range wh.warehouses {
+			source := warehouse.Source
+			destination := warehouse.Destination
+
+			if !source.Enabled || !destination.Enabled {
+				continue
+			}
+
+			config := destination.Config
+			// Default frequency
+			syncFrequency := "1440"
+			if config[warehouseutils.SyncFrequency] != nil {
+				syncFrequency, _ = config[warehouseutils.SyncFrequency].(string)
+			}
+
+			timeWindow := uploadBufferTimeInMin
+			if value, err := strconv.Atoi(syncFrequency); err == nil {
+				timeWindow += value
+			}
+
+			sqlStatement := fmt.Sprintf(`
+				select 
+					cast( case when count(*) > 0 then 1 else 0 end as bit ), 
+						(select count(*) from %[7]s where source_id='%[2]s' and destination_id='%[3]s' and created_at > now() - interval '%[6]d MIN') as staging_files_count 
```

Comment:
let's do the staging file count query separately first and then do the uploads query if there are staging files? Also, for staging files let us do something like `select created_at from wh_staging_files order by created_at desc limit 1` and compare it with time window. This might be faster than filtering on time window in query

Reply:
done